packages/workshop-app/app/utils/mdx.tsx

```diff
@@ -6,6 +7,35 @@ import { LaunchEditor } from '~/routes/launch-editor'
 import { AnchorOrLink } from './misc'
 import Accordion from '~/components/accordion'
 
+function getCode(data: any) {
+	// just in case we are lost in space
+	try {
+		if (typeof data === 'string') return data
+		const { children } = data.props
+		if (typeof children === 'string') return children
+		return children.map(getCode).flat().join('')
+	} catch {}
+	return null
+}
+
+export function preWithCopyToClipboard({ children, ...props }: any) {
```

Comment:
Let's make this capital because it's a component:

```suggestion
export function PreWithCopyToClipboard({ children, ...props }: any) {
```

Reply:
Done